File: app/src/main/java/com/hosigus/askposts/item/jsonBean/BaseBean.java
package com.hosigus.askposts.item.jsonBean;

import com.hosigus.tools.items.JSONBean;

import org.json.JSONException;
import org.json.JSONObject;

/**
 * Created by 某只机智 on 2018/5/25.
 * "status": 200,
 * "info": "success",
 * "data": null
 */
public class BaseBean extends JSONBean {
    private int status;
    private String info;
    private String data;
    @Override
    public void parser(JSONObject object) {
        try {
            status = object.getInt("status");
            info = object.getString("info");
            data = isStatusOK() && object.has("data") ? object.getString("data") : null;
        } catch (JSONException e) {
            e.printStackTrace();
        }
    }

    public int getStatus() {
        return status;
    }

    public String getInfo() {
        return info;
    }

    public String getData() {
        return data;
    }

    public boolean isStatusOK() {
        return status == 200;
    }

    public boolean isGoodJson() {
        return isStatusOK() && data != null && !data.isEmpty();
    }
}

File: app/src/main/java/com/hosigus/askposts/item/jsonBean/BaseBean.kt
package com.hosigus.askposts.item.jsonBean

import com.hosigus.tools.items.JSONBean
import org.json.JSONObject


/**
 * Created by Hosigus on 2018/5/30.
 * "status": 200,
 * "info": "success",
 * "data": null
 */
class BaseBean : JSONBean() {
    var status:Int = 0
        private set
    lateinit var info: String
        private set
    var data: String? = null
        private set

    override fun parser(json: JSONObject) {
        status = json.getInt("status")
        info = json.getString("info")
        data = if(isStatusOK()&&json.has("data")) json.getString("data") else null
    }
    
    fun isStatusOK() = status == 200
    fun isGoodJson() = isStatusOK() && !data.isNullOrBlank()

}